odoo/orm/registry.py

```diff
@@ -139,7 +159,15 @@ def new(cls, db_name: str, force_demo: bool = False, status: None = None, update
             # This should be a method on Registry
             from odoo.modules.loading import load_modules, reset_modules_state  # noqa: PLC0415
             try:
-                load_modules(registry, force_demo, update_module=update_module)
+                if new_db_demo is None:
+                    new_db_demo = not config['without_demo']
+                load_modules(
+                    registry,
+                    update_module=update_module,
```

Comment:
Maybe we could have the `update_module` optionally given by updating this line.
```suggestion
                    update_module=update_module or bool(upgrade_modules or install_modules),
```